crates/interfaces/src/p2p/error.rs

```diff
@@ -11,14 +11,14 @@ pub type RequestResult<T> = Result<T, RequestError>;
 /// Result with [PeerId]
 pub type PeerRequestResult<T> = RequestResult<WithPeerId<T>>;
 
-/// Trait used to validate requests
-pub trait RequestValidation {
-    /// Determine whether the response matches what we requested in request
-    fn is_likely_a_bad_message(&self, request: &HeadersRequest) -> bool;
+/// Helper trait used to validate responses.
+pub trait EthResponseValidator {
+    /// Determine whether the response matches what we requested in [HeadersRequest]
+    fn is_likely_a_bad_headers_response(&self, request: &HeadersRequest) -> bool;
 }
 
-impl RequestValidation for RequestResult<Vec<Header>> {
-    fn is_likely_a_bad_message(&self, request: &HeadersRequest) -> bool {
+impl EthResponseValidator for RequestResult<Vec<Header>> {
+    fn is_likely_a_bad_headers_response(&self, request: &HeadersRequest) -> bool {
```

Comment:
the trait method seems a bit verbose. also, since it accepts `HeadersRequest`, i assume, it will only be used for headers validation, but can be implemented for different protocols. 
```suggestion
impl HeadersResponseValidator for RequestResult<Vec<Header>> {
    fn is_bad_response(&self, request: &HeadersRequest) -> bool {
```

Reply:
we can add another function for bodies as well, hence the rename